File: src/Game/Payload/FxEvent.ts
export type FxEvent =
  | { type: "BallWallHit"; intensity: number }
  | { type: "GoalScored"; team_id: number };

export function normalizeFxEvent(e: any): FxEvent {
  const key = Object.keys(e)[0];

  switch (key) {
    case "GoalScored":
      return { type: "GoalScored", ...e[key] };

    case "BallWallHit":
      return { type: "BallWallHit", ...e[key] };

    default:
      throw new Error("Unknown FxEvent: " + key);
  }
}

File: src-tauri/src/game/frontend/fxevent.rs
#[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
pub enum FxEvent {
    BallWallHit { pos: (f32, f32), intensity: f32 },
    GoalScored { team_id: u8 },
    // PlaySound {
    //     kind: SoundKind,
    //     pos: (f32, f32),
    // },
}
